src/sync_std.rs

```diff
@@ -36,6 +36,16 @@ impl Condvar {
         }
     }
 
+    pub fn wait_timeout<'a, T>(
+        &self,
+        guard: MutexGuard<'a, T>,
+        duration: Duration,
+    ) -> Option<MutexGuard<'a, T>> {
+        match self.inner.wait_timeout(guard, duration) {
```

Comment:
I was able to spot it hang indefinitely (with `duration` set to 1 second) - tested with `dbg!()` before and after and I have no idea why. The problem disappears when switching to `parking_lot`.